Combine various webhook content pages (#1105)
mattermost/mattermost-developer-documentation: site/content/integrate/webhooks/incoming/_index.md

title: Incoming webhooks
description: Incoming webhooks
weight: 10
aliases:
  - /integrate/webhooks/incoming/incoming-webhooks/
  - /integrate/webhooks/incoming/using-incoming-webhooks/
---
## Create an incoming webhook

Let's learn how to create a simple incoming webhook that posts the following message to Mattermost.

![image](incoming_webhooks_create_simple.png)

> 1. First, go to **Product menu > Integrations > Incoming Webhook**.

Reviewer: ```suggestion
1. In Mattermost, go to **Product menu > Integrations > Incoming Webhook**.
```

>     - If you don't have the **Integrations** option in your Main Menu, incoming webhooks may not be enabled on your Mattermost server or may be disabled for non-admins. They can be enabled by a System Admin from **System Console > Integrations > Integration Management**. Then continue with the steps below.

Reviewer: ```suggestion
    - If you don't have the **Integrations** option, incoming webhooks may not be enabled on your Mattermost server or may be disabled for non-admins. They can be enabled by a System Admin from **System Console > Integrations > Integration Management**. Once incoming webhooks are enabled, continue with the steps below.
```

> 2. Select **Add Incoming Webhook** and add name and description for the webhook. The description can be up to 500 characters.

Reviewer: ```suggestion
2. Select **Add Incoming Webhook** and a add name and description for the webhook. The description can be up to 500 characters.
```

Author: It looks like the `a` should come after the `add`.

Reviewer: Thanks! Suggestion updated!

```

__Treat this endpoint as a secret.__ Anyone who has it will be able to post messages to your Mattermost instance.

## Use an incoming webhook

To use the endpoint, have your application make the following request:

```http request
POST /hooks/xxx-generatedkey-xxx HTTP/1.1
Host: your-mattermost-server.com
Content-Type: application/json
Content-Length: 63

{"text": "Hello, this is some text\nThis is more text. :tada:"}
```

For example, here is the same request using `cURL`:

```bash
curl -i -X POST -H 'Content-Type: application/json' -d '{"text": "Hello, this is some text\nThis is more text. :tada:"}' https://your-mattermost-server.com/hooks/xxx-generatedkey-xxx
```

For compatibility with Slack incoming webhooks, if no `Content-Type` header is set then the request body must be prefixed with `payload=`, like so:

```
payload={"text": "Hello, this is some text\nThis is more text. :tada:"}
```

A successful request will get the following response:

```
HTTP/1.1 200 OK
Content-Type: text/plain
X-Request-Id: hoan6o9ws7rp5xj7wu9rmysrte
X-Version-Id: 4.7.1.dev.12799cd77e172e8a2eba0f9091ec1471.false
Date: Sun, 04 Mar 2018 17:19:09 GMT
Content-Length: 2

ok
```

> All webhook posts will display a `BOT` indicator next to the username in Mattermost clients, to help prevent against [phishing attacks](https://en.wikipedia.org/wiki/Phishing).

Reviewer: ```suggestion
All webhook posts will display a `BOT` indicator next to the username in Mattermost clients to help prevent against [phishing attacks](https://en.wikipedia.org/wiki/Phishing).
```
